### PR TITLE
modified .import to .cache for project files

### PR DESCRIPTION
**Pull request by ERUD1T3**
Fixes ambiguity raised in issue #23368. However, it is not yet backward compatible. I will be working on making it able to recognize both .cache and .import folders in the user-created project directory.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Visual scripting.
 - Rewritten import system.
   - Assets are now referenced with their source files, then imported in a transparent manner by the engine.
-  - Imported assets are now cached in a `.import` directory, making distribution and versioning easier.
+  - Imported assets are now cached in a `.cache` directory, making distribution and versioning easier.
```
Comment (LikeLakers2) [inline CHANGELOG.md]: We shouldn't be editing the 3.0 changelog. 3.0's already been released.

**Comment by ERUD1T3** [comment]
added F8 shortcut to activate/deactivate plugin

**Comment by vnen** [comment]
I don't like the name `.cache`, because it does not describe what the folder really is. I would call it `.imported` or something.

**Comment by Zireael07** [comment]
@vnen: .imported and the like means back to the ambiguity this means to solve.

**Comment by asheraryam** [comment]
The names `.import_cache` or `.importcache ` might be great.

**Comment by akien-mga** [comment]
Personally I don't think it's worth breaking compat for what is IMO not a big issue. Maybe for 4.0, but not for 3.1 nor 3.2.

**Comment by akien-mga** [comment]
This PR also has completely unrelated commits that should be moved to their own PRs.

**Comment by ERUD1T3** [comment]
Sorry for the combined PRs, still figuring out pull request mechanics. I will separate them as asked. I will also try to change to .import_cache. Thanks again for the feedback

**Comment by vnen** [comment]
> .imported and the like means back to the ambiguity this means to solve.

I disagree. The issue is that the names are exactly the same. If they are different, even by a small amount, it's clearer which one are you talking about when discussing. Also not confusing for `.gitignore`.

**Comment by aaronfranke** [comment]
I don't see why `.cache` is an unsuitable name. These files are regeneratable without problems from other existing files, they are basically Godot's working cache of them.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,7 +59,8 @@ by drag and dropping the file in the GitHub edition field.
 
 If you want to add new engine functionalities, please make sure that:
 
-* This functionality is desired.
+* This functionality is desired, which means that it solves a common use case
+  that several users will need in their real-life projects.
```
Comment (aaronfranke) [inline CONTRIBUTING.md:63]: These are unrelated changes and shouldn't be in this PR.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

**Comment by aaronfranke** [comment]
Closing because:

* Git history is a mess.

* No response from the author for over a year.

* Superceded by #38607

**Pull request closed without merging**